### PR TITLE
feat: endpoint to fetch transaction status of an order via paytm

### LINKED ISSUE #6401: Implement route to get transaction status of the payment
**Is your feature request related to a problem? Please describe.**

Parent Issue - #6247 
Implement route to get transaction status of the payment to verify if payment is actually made or not

**Additional context**

On it

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #6401 

#### Short description of what this resolves:
Implement route to get transaction status of the payment to verify if payment is actually made or not

#### Changes proposed in this pull request:
- adds a route to check the transaction status

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by shreyanshdwivedi** [comment]
This PR can be merged after https://github.com/fossasia/open-event-server/pull/6399 is merged.
The hound errors will automatically be resolved once the mentioned PR is merged

**Comment by shreyanshdwivedi** [comment]
@uds5501 @mrsaicharan1 please review this

**Comment by shreyanshdwivedi** [comment]
@mrsaicharan1 please review

File: app/api/orders.py [hunk 1]
```diff
@@ -766,3 +766,30 @@ def process_transaction(order_identifier, txn_token):
 
     response = PaytmPaymentsManager.hit_paytm_endpoint(url=url, head=head, body=body)
     return response
+
+
+@order_misc_routes.route('/orders/<string:order_identifier>/paytm/transaction-status', methods=['GET'])
+def get_transaction_status(order_identifier):
+    paytm_params = dict()
+    paytm_checksum_params = dict()
+    url = ""
```
Comment (mrsaicharan1) [inline app/api/orders.py]: Include a check alluding to the target endpoint(based on `paytm_mode` in settings)
_reference_:
Staging:
https://securegw-stage.paytm.in/order/status

Production:
https://securegw.paytm.in/order/status

Comment (shreyanshdwivedi) [reply app/api/orders.py]: I've already used it on line 789 @mrsaicharan1

**Comment by shreyanshdwivedi** [comment]
@mrsaicharan1 I cannot use the helper `hit_paytm_endpoint` because the `Transaction Status API` doesn't ask us to segregate the paytmParams in head and body but your helper does so.
This API is different from others. Please take a look at https://developer.paytm.com/docs/transaction-status-api/

File: app/api/orders.py (hunk 1 above)
Comment (mrsaicharan1) [reply app/api/orders.py]: @shreyanshdwivedi I guess that's fine then. Remove this explicit declaration of `url=""` and this is GTG. :D

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal please review

**Pull request merged**